jax/experimental/maps.py

```diff
@@ -53,27 +53,18 @@
 
 EXPERIMENTAL_SPMD_LOWERING = False
 
-class FrozenDict:  # dataclasses might remove some boilerplate here
+class FrozenDict(abc.Mapping):
   def __init__(self, *args, **kwargs):
     self.contents = dict(*args, **kwargs)
 
-  allowed_methods = {'items', 'values', 'keys', 'get'}
-  def __getattr__(self, name):
-    if name in self.allowed_methods:
-      return getattr(self.contents, name)
-    raise AttributeError(name)
-
   def __iter__(self):
-    return self.contents.__iter__()
+    return iter(self.contents)
 
   def __len__(self):
-    return self.contents.__len__()
+    return len(self.contents)
 
   def __getitem__(self, name):
-    return self.contents.__getitem__(name)
-
-  def __eq__(self, other):
-    return isinstance(other, FrozenDict) and self.contents == other.contents
+    return self.contents[name]
```

Comment:
Good catch, sorry about that. I thought it would be save to use the [default implementation of `abc.Mapping`](https://github.com/python/cpython/blob/37a9d43dd2a7f2b73f60c43a94a1229d6de5d674/Lib/_collections_abc.py#L786-L789) here. I restored the previous behaviour.